### PR TITLE
Vault-2713: Retry Irrevocable Lease Revocation Periodically

### PR DESCRIPTION
**Pull request by swayne275**

### CODE REVIEW

File: vault/expiration.go [hunk 1]
```diff
@@ -943,6 +956,27 @@ func (m *ExpirationManager) lazyRevokeInternal(ctx context.Context, leaseID stri
...
 }
 
+// TODO should context passed to revokeCommon contain the lease namespace?
+// should be run on a schedule. something like once a day, maybe once a week
+func (m *ExpirationManager) attemptIrrevocableLeasesRevoke() {
+	m.irrevocable.Range(func(k, v interface{}) bool {
+		leaseID := k.(string)
+		le := v.(*leaseEntry)
+
+		if le.ExpireTime.Add(7 * 24 * time.Hour).Before(time.Now()) {
```
Comment (briankassouf) [inline vault/expiration.go:966]: I think this could be reduced quite a bit, what about just an hour ago?

File: vault/expiration.go [hunk 2]
```diff
@@ -943,6 +956,27 @@ func (m *ExpirationManager) lazyRevokeInternal(ctx context.Context, leaseID stri
...
+			// if the lease has expired at least a week ago, try again
+			if err := m.revokeCommon(context.Background(), leaseID, false, false); err != nil {
```
Comment (briankassouf) [inline vault/expiration.go:968]: Can we derive a context from a.activeContext and add a timeout here?

**Review by sgmiller — APPROVED** [review]
Agree with Brian's recs but otherwise lgtm.

File: vault/expiration.go (hunk 2 above)
Comment (swayne275) [reply vault/expiration.go:968]: how much timeout would you recommend? 1 minute?

Comment (swayne275) [reply vault/expiration.go]: 6197b3c04

let me know what you think about the 1 minute timeout

File: vault/expiration.go (hunk 1 above)
Comment (swayne275) [reply vault/expiration.go]: 6197b3c04

**Review by briankassouf — APPROVED** [review]

File: vault/expiration.go (hunk 2 above)
Comment (sgmiller) [reply vault/expiration.go]: I think 1m is fine since this is the retry code.

**Pull request merged**